### PR TITLE
make inner classes static where possible

### PR DESCRIPTION
**Pull request by stevencrockett**
First commit to gradle. 🍻 Starting simple.

I ran static analysis on the codebase which flagged all these classes. They are non-static, but have no need to be. 

Static innner classes are conceptually simpler, being equivalent to normal top-level classes in Java. Non-static inner class instances will contain a reference to the enclosing instance.

So making these static makes the code a bit simpler, and also will save a bit on memory.

### CODE REVIEW

**Comment by stevencrockett** [comment]
So the build blew up. I'm assuming the jvm doesn't have enough memory.
 
```
* What went wrong:
Execution failed for task ':toolingApi:compileCrossVersionTestGroovy'.
> Java heap space
```

I'm looking at other PRs though, and I have no idea why I get a travis build where teamcity seems to be used elsewhere. The jobs may be configured differently.

**Comment by big-guy** [comment]
Thanks @stevencrockett - I kicked off a full pipeline build on our TeamCity instance.

The TravisCI build is just something we use to let outside contributors see _some_ feedback before a maintainer looks at the PR.  It looks like in this case, the build was a bit too much for the TravisCI builder.

**Comment by stevencrockett** [comment]
@big-guy Thanks a lot. There are a couple of issues I need to look into, so no point reviewing yet.

Are there preferences with PR workflow for Gradle? i.e. Am I fine to amend my commit and rebase? Should I be adding incremental commits and squash at the end?

File: subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperGenerationIntegrationTest.groovy [hunk 1]
```diff
@@ -71,7 +71,7 @@ class WrapperGenerationIntegrationTest extends AbstractIntegrationSpec {
         executer.inDirectory(file("second")).withTasks("wrapper").run()
 
         then: "the checksum should be constant (unless there are code changes)"
-        sha256(file("first/gradle/wrapper/gradle-wrapper.jar")).asHexString() == "3888c76faa032ea8394b8a54e04ce2227ab1f4be64f65d450f8509fe112d38ce"
+        sha256(file("first/gradle/wrapper/gradle-wrapper.jar")).asHexString() == "96f793a18e056c23ffeec67c1f3bb8eccff5a4a407fc9ceac183527e7eedf4b6"
```
Comment (stevencrockett) [inline subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperGenerationIntegrationTest.groovy]: So I rebased and I had to undo a couple of the changes as they were test classes where tests were actually relying on them being non-static.

Other than that, this is the only other notable change. Given I've touched the `org.gradle.wrapper.PathAssembler` class, the bytecode is different and so this hash must've changed.

**Comment by stevencrockett** [comment]
@big-guy I've updated things, and this lightweight travis build has passed at least. I'm assuming I'm dependent on a maintainer to kick-off a full TC build.

**Comment by rieske** [comment]
Thank you @stevencrockett!

**Pull request merged**